src/crawler.ts

```diff
@@ -1582,27 +1587,18 @@ self.__bx_behaviors.selectMainBehavior();
 
       // Handle 4xx or 5xx response as a page load error
       const statusCode = resp.status();
-      const statusString = statusCode.toString();
-      if (
-        statusString.startsWith("4") ||
-        statusString.startsWith("5") ||
-        isChromeError
-      ) {
+      data.statusCode = statusCode;
```

Comment:
I think we need to add this for directly fetched non-HTML pages above as well to avoid the status code of 0 for fetched images, videos, etc.